docs/standard/garbage-collection/implementing-dispose.md

```diff
@@ -1,158 +1,173 @@
+
 > [!NOTE]
-> In C#, you override <xref:System.Object.Finalize%2A?displayProperty=nameWithType> by defining a [destructor](../../csharp/programming-guide/classes-and-structs/destructors.md).  
-  
+> In C#, you override <xref:System.Object.Finalize%2A?displayProperty=nameWithType> by defining a [destructor](../../csharp/programming-guide/classes-and-structs/destructors.md). In VB, this is done with `Protected Overrides Sub Finalize()`.
+
 <a name="SafeHandles"></a>
 ## Using safe handles
 
-Writing code for an object's finalizer is a complex task that can cause problems if not done correctly. Therefore, we recommend that you construct <xref:System.Runtime.InteropServices.SafeHandle?displayProperty=nameWithType> objects instead of implementing a finalizer.  
-  
-Classes derived from the <xref:System.Runtime.InteropServices.SafeHandle?displayProperty=nameWithType> class simplify object lifetime issues by assigning and releasing handles without interruption. They contain a critical finalizer that is guaranteed to run while an application domain is unloading. For more information about the advantages of using a safe handle, see <xref:System.Runtime.InteropServices.SafeHandle?displayProperty=nameWithType>. The following derived classes in the <xref:Microsoft.Win32.SafeHandles> namespace provide safe handles:  
-  
-- The <xref:Microsoft.Win32.SafeHandles.SafeFileHandle>, <xref:Microsoft.Win32.SafeHandles.SafeMemoryMappedFileHandle>, and <xref:Microsoft.Win32.SafeHandles.SafePipeHandle> class, for files, memory mapped files, and pipes.  
-  
-- The <xref:Microsoft.Win32.SafeHandles.SafeMemoryMappedViewHandle> class, for memory views.  
-  
-- The <xref:Microsoft.Win32.SafeHandles.SafeNCryptKeyHandle>, <xref:Microsoft.Win32.SafeHandles.SafeNCryptProviderHandle>, and <xref:Microsoft.Win32.SafeHandles.SafeNCryptSecretHandle> classes, for cryptography constructs.  
-  
-- The <xref:Microsoft.Win32.SafeHandles.SafeRegistryHandle> class, for registry keys.  
-  
-- The <xref:Microsoft.Win32.SafeHandles.SafeWaitHandle> class, for wait handles.  
-  
+Writing code for an object's finalizer is a complex task that can cause problems if not done correctly. Therefore, we recommend that you construct <xref:System.Runtime.InteropServices.SafeHandle?displayProperty=nameWithType> objects instead of implementing a finalizer.
+
+Classes derived from the <xref:System.Runtime.InteropServices.SafeHandle?displayProperty=nameWithType> class simplify object lifetime issues by assigning and releasing handles without interruption. They contain a critical finalizer that is guaranteed to run while an application domain is unloading. For more information about the advantages of using a safe handle, see <xref:System.Runtime.InteropServices.SafeHandle?displayProperty=nameWithType>. The following derived classes in the <xref:Microsoft.Win32.SafeHandles> namespace provide safe handles:
```

Comment:
All of the reliability discussion here isn't relevant to .NET Core.  I'd suggest changing this to something like:

SafeHandle is an abstract managed type.  It wraps an IntPtr that identifies an unmanaged resource, generally a handle on Windows or a file descriptor on Unix, and it provides all of the logic necessary to ensure that this handle is released once and only once, when the SafeHandle is disposed of or when all references to the SafeHandle have been dropped and the SafeHandle instance is finalized.  Types derive from SafeHandle to override cleanup logic specific to a particular kind of handle, namely what values for the IntPtr are considered invalid and how to actually free the handle.  For example, SafeFileHandle derives from SafeHandle to wrap IntPtrs that identify open file handles/descriptors, and overrides its ReleaseHandle method to close it (via the `close` function on Unix or `CloseHandle` function on Windows).  Most APIs in .NET libraries that create an unmanaged resource will wrap it in a SafeHandle and return that SafeHandle to you as needed, rather than handing back the raw pointer.  In situations where you yourself interact with an unmanaged component and get an IntPtr for an unmanaged resource, you can create your own SafeHandle type to wrap it.  As a result, very few non-SafeHandle types need to implement finalizers; most disposable pattern implementations only end up wrapping other managed resources, some of which may be SafeHandles.